hooks/useAPI.ts

```diff
@@ -0,0 +1,65 @@
+import { SALEOR_DOMAIN_HEADER } from "@saleor/app-sdk/const";
+import { useEffect, useState } from "react";
+import useApp from "./useApp";
+
+type Options = Record<string, string>;
+
+interface UseFetchProps {
+  url: string;
+  options?: Options;
+  skip?: boolean;
+}
+
+// This hook is meant to be used mainly for internal API calls
```

Comment:
I would consider a different name. Since the app can communicate with the Saleor, external API and its own API it may not be clear which one we are targeting. WDYT about `useAppApi`?